src/Evolution/Systems/GrMhd/GhValenciaDivClean/BoundaryConditions/DirichletFreeOutflow.cpp

```diff
@@ -8,18 +8,265 @@
+          gr::Tags::SpacetimeMetric<DataVector, 3>,
+          ::gh::Tags::Pi<DataVector, 3>, ::gh::Tags::Phi<DataVector, 3>>,
+      ghmhd::GhValenciaDivClean::InitialData::analytic_solutions_and_data_list>(
+      analytic_prescription_.get(),
+      [&coords, &time](const auto* const initial_data) {
+        using spacetime_tags = tmpl::list<
+            gr::Tags::SpatialMetric<DataVector, 3>,
+            gr::Tags::InverseSpatialMetric<DataVector, 3>,
+            gr::Tags::SqrtDetSpatialMetric<DataVector>,
+            gr::Tags::Lapse<DataVector>, gr::Tags::Shift<DataVector, 3>,
+            gr::Tags::SpacetimeMetric<DataVector, 3>,
+            ::gh::Tags::Pi<DataVector, 3>, ::gh::Tags::Phi<DataVector, 3>>;
+        if constexpr (is_analytic_solution_v<
+                          std::decay_t<decltype(*initial_data)>>) {
+          return initial_data->variables(coords, time, spacetime_tags{});
+        } else if constexpr (evolution::is_numeric_initial_data_v<
+                                 std::decay_t<decltype(*initial_data)>>) {
+          ERROR(
+              "Cannot currently use numeric initial data as an analytic "
+              "prescription for boundary conditions.");
+        } else {
+          (void)time;
+          return initial_data->variables(coords, spacetime_tags{});
+        }
+      });
+
+  *spacetime_metric =
+      get<gr::Tags::SpacetimeMetric<DataVector, 3>>(boundary_values);
+  *pi = get<::gh::Tags::Pi<DataVector, 3>>(boundary_values);
+  *phi = get<::gh::Tags::Phi<DataVector, 3>>(boundary_values);
+  *lapse = get<gr::Tags::Lapse<DataVector>>(boundary_values);
+  *shift = get<gr::Tags::Shift<DataVector, 3>>(boundary_values);
+  *inv_spatial_metric =
+      get<gr::Tags::InverseSpatialMetric<DataVector, 3>>(boundary_values);
+
+  return grmhd::ValenciaDivClean::BoundaryConditions::HydroFreeOutflow::
+      dg_ghost(tilde_d, tilde_ye, tilde_tau, tilde_s, tilde_b, tilde_phi,
+
+               tilde_d_flux, tilde_ye_flux, tilde_tau_flux, tilde_s_flux,
+               tilde_b_flux, tilde_phi_flux,
+
+               lapse, shift, inv_spatial_metric,
+
+               face_mesh_velocity, normal_covector, normal_vector,
+
+               interior_rest_mass_density, interior_electron_fraction,
+               interior_specific_internal_energy, interior_spatial_velocity,
+               interior_magnetic_field, interior_lorentz_factor,
+               interior_pressure,
+
+               *shift, *lapse, *inv_spatial_metric);
+}
+
+void DirichletFreeOutflow::fd_ghost(
+    const gsl::not_null<tnsr::aa<DataVector, 3, Frame::Inertial>*>
+        spacetime_metric,
+    const gsl::not_null<tnsr::aa<DataVector, 3, Frame::Inertial>*> pi,
+    const gsl::not_null<tnsr::iaa<DataVector, 3, Frame::Inertial>*> phi,
+    const gsl::not_null<Scalar<DataVector>*> rest_mass_density,
+    const gsl::not_null<Scalar<DataVector>*> electron_fraction,
+    const gsl::not_null<Scalar<DataVector>*> pressure,
```

Comment:
should this also be temperature now?

Reply:
I'm not sure, the hydro variables get passed to HydroFreeOutflow so maybe @yoonso0-0 would know?

Reply:
Ah it certainly looks like it [wants to be ](https://github.com/sxs-collaboration/spectre/blob/6b5d00a2c878f6a1472190b8a33bcaddd3a1f69f/tests/Unit/Evolution/Systems/GrMhd/GhValenciaDivClean/BoundaryConditions/Test_DirichletFreeOutflow.cpp#L437)

Reply:
Yeah I think it looks like there's a few different things that need to be changed, I think this should probably be a separate bug fix PR.  For example it's not clear what the inputs to`  grmhd::ValenciaDivClean::ConservativeFromPrimitive::apply(...` should be